spec/mustache.spec.ts

```diff
@@ -28,38 +28,25 @@ describe("fromRemoteMustacheTemplate", () => {
   hooks(libraryScript)
 
   it("fetches remote templates url", async () => {
-    const openSpy = jest.spyOn(XMLHttpRequest.prototype, "open")
-    const sendSpy = jest.spyOn(XMLHttpRequest.prototype, "send")
-
     const mockUrl = "template.mustache"
     const render = fromRemoteMustacheTemplate(mockUrl)
 
-    const mockXhr = {
-      open: jest.fn(),
-      send: jest.fn(),
-      status: 200,
-      responseText: mustacheTemplate,
-      onload: jest.fn(),
-      onerror: jest.fn(),
-    }
-
-    openSpy.mockImplementation((method, url) => {
+    // @ts-expect-error partial mock
+    global.fetch = jest.fn(url => {
       if (url === mockUrl) {
-        return mockXhr.open(method, url)
+        return Promise.resolve({
```

Comment:
The shape looks similar with liquid and mustache specs. Would it be worth to extract it?